service/debugger/debugger.go

```diff
@@ -768,20 +796,61 @@ func (d *Debugger) ClearBreakpoint(requestedBp *api.Breakpoint) (*api.Breakpoint
 	}
 
 	clearedBp := api.ConvertBreakpoints(bps)
+	clearedBp = append(clearedBp, dbps...)
 	if len(clearedBp) < 0 {
 		return nil, nil
 	}
 	d.log.Infof("cleared breakpoint: %#v", clearedBp)
 	return clearedBp[0], nil
 }
 
+// ToggleBreakpoint toggles on or off a breakpoint.
+func (d *Debugger) ToggleBreakpoint(requestedBp *api.Breakpoint) (*api.Breakpoint, error) {
+	toggle := func(addr uint64) {
+		if dbp, ok := d.disabledBreakpoints[addr]; ok {
+			if bp, err := d.target.SetBreakpointWithID(dbp.ID, dbp.Addr); err == nil {
```

Comment:
If this is successful you should call `copyBreakpointInfo(bp, dbp)`, otherwise the new breakpoint won't have the correct attributes.